test/functional/invalidtxrequest.py

```diff
@@ -8,55 +8,46 @@
 import time
 
 from test_framework.blocktools import create_block, create_coinbase, create_transaction
-from test_framework.comptool import RejectResult, TestInstance, TestManager
 from test_framework.messages import COIN
-from test_framework.mininode import network_thread_start
-from test_framework.test_framework import ComparisonTestFramework
+from test_framework.mininode import network_thread_start, P2PStub
+from test_framework.test_framework import BitcoinTestFramework
 
-class InvalidTxRequestTest(ComparisonTestFramework):
+class InvalidTxRequestTest(BitcoinTestFramework):
 
     def set_test_params(self):
         self.num_nodes = 1
         self.setup_clean_chain = True
+        self.extra_args = [["-whitelist=127.0.0.1"]]
 
     def run_test(self):
-        test = TestManager(self, self.options.tmpdir)
-        test.add_all_connections(self.nodes)
-        self.tip = None
-        self.block_time = None
+        # Add p2p connection to node0
+        node = self.nodes[0]  # convenience reference to the node
+        node.add_p2p_connection(P2PStub())
+
         network_thread_start()
-        test.run()
+        node.p2p.wait_for_verack()
 
-    def get_tests(self):
-        self.tip = int("0x" + self.nodes[0].getbestblockhash(), 0)
-        self.block_time = int(time.time()) + 1
+        tip = int("0x" + self.nodes[0].getbestblockhash(), 0)
+        block_time = int(time.time()) + 1
```

Comment:
In commit "Change invalidtxrequest to use BitcoinTestFramework"

Could maybe switch from time.time() to best_block["time"] to be consistent with invalidblockrequest change in #11172. It seems better in a test environment, to be using information from the environment instead of calling out to the OS.

Reply:
Yes, good idea